### PR TITLE
[docs] `Stack` layout guide improvements

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Currently, the `Stack` layout requires improvement in form of content structure, aligning sections with a section title, and other inconsistencies with code blocks, typos, and grammatical errors.

# How

This PR updates the `Stack` layout guide by adding a Get started section to make it easier for developers to learn how to set up their app with file-based routing to create a `stack` navigation layout.

It also includes verbiage improvements, consistent language and terms throughout the guide, fixing grammatical issues and typos, improving code snippets and adding highlights to focus on relevant info, and improving the overall guide's structure.

# Test Plan

CleanShot 2024-04-24 at 17 57 53@2x

Preview: /router/advanced/stack/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by jonsamp — APPROVED** [review]
Sorry for the late review! I reviewed this and forgot to hit submit. I have a few things I think we could change but overall this is an improvement. Once you've made those changes, I think we should merge it!

File: docs/pages/router/advanced/stack.mdx [hunk 1]
```diff
@@ -1,31 +1,94 @@
 ---
 title: Stack
-description: Learn how to use the Stack Layout in Expo Router.
+description: Learn how to use the Stack layout in Expo Router.
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/router/advanced/stack.mdx:3]: I've updated this to "navigator" and will also follow up to use the same term in other guides that are mentioned in your comment.

File: docs/pages/router/advanced/stack.mdx [hunk 2]
```diff
@@ -1,31 +1,94 @@
...
+```tsx app/_layout.tsx
 import { Stack } from 'expo-router/stack';
 
 export default function Layout() {
   return <Stack />;
 }
 ```
 
-## Configure header bar
+> **info** The `Stack` layout in Expo Router wraps the [Native Stack Navigator](https://reactnavigation.org/docs/native-stack-navigator) from React Navigation (not to be confused with the legacy [JS Stack Navigator](https://reactnavigation.org/docs/stack-navigator)).
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/router/advanced/stack.mdx]: Moved into a different section at the bottom of the page.

CleanShot 2024-04-24 at 17 56 44@2x

**Comment by amandeepmittal** [comment]
@jonsamp, I've applied all of your feeback. Will follow up on this for other guides as well in a separate PR:

> ... The other types of routing methods (tab, drawer, modal) I think we'd say "navigator" after each.

**Pull request merged**